sharedbuffers/mapped_struct.py

```diff
@@ -812,14 +819,26 @@ def unpack_from(cls, buf, offs, idmap = None):
         buf = _likerobuffer(buf)
         return cls(buf, offs, idmap)
 
-    @cython.locals(obj_offs = cython.ulonglong, dcode = cython.char, index = cython.ulonglong, pindex = "unsigned long *", dataoffs = cython.ulonglong)
-    def __getitem__(self, index):
+
+    @cython.locals(obj_offs = cython.ulonglong, dcode = cython.char, index = cython.longlong, pindex = "unsigned long *",
+        dataoffs = cython.ulonglong, xlen = cython.ulonglong)
+    def _getitem(self, index):
 
         dcode, objlen, itemsize, dataoffs, _struct = self._metadata()
+        xlen = objlen
+
+        if self.elem_step != 0:
+            xlen = self.elem_end - self.elem_start
+            index *= self.elem_step
 
-        if index >= objlen:
+        if index < 0:
+            index += xlen
+
+        if index >= xlen or index < 0 or index >= objlen:
```

Comment:
I think that checking for `index >= objlen` is unnecessary.

Reply:
Yeah, @llogiudice said the same, but I'd rather have the check for safety. It's the only thing that clearly and inequivocally avoids a buffer overflow.